eng/tools/sdk-suppressions/src/updateSdkSuppressionsLabel.ts

```diff
@@ -0,0 +1,318 @@
+import _ from "lodash";
+import { writeFileSync } from "fs";
+import { sdkLabels, SdkName } from "./sdk.js";
+import {
+  SdkSuppressionsYml,
+  SdkSuppressionsSection,
+  sdkSuppressionsFileName,
+  SdkPackageSuppressionsEntry,
+  validateSdkSuppressionsFile,
+} from "./sdkSuppressions.js";
+import { parseYamlContent, runGitCommand } from "./common.js";
+
+/**
+ *
+ * @param pr
+ * @returns SdkName list
+ * This part compares the suppression files of the head branch and the base branch.
+ * To get the SDK, we need to identify which package name, SDK name, or breaking changes are different and apply the SDK suppression label accordingly in the next step.
+ * change details can see at getSdkNamesWithChangedSuppressions function
+ * on the other hand that the sdkName list will return an empty array if it does not have a suppression file or if the file is blank.
+ */
+export async function getSdkSuppressionsSdkNames(
+  prChangeFiles: string,
+  baseCommitHash: string,
+  headCommitHash: string
+): Promise<SdkName[]> {
+  console.log(`Will compare base commit: ${baseCommitHash} and head commit: ${headCommitHash} to get different SDK.`);
+  const filesChangedPaths = prChangeFiles.split(" ");
+  console.log(`The pr origin changed files: ${filesChangedPaths.join(", ")}`);
+  let suppressionFileList = filterSuppressionList(filesChangedPaths);
+  console.log(`Will compare sdk-suppression.yaml files: ${suppressionFileList.join(", ")}`);
+  let sdkNameList: SdkName[] = [];
+  if (suppressionFileList.length > 0) {
+    for (const suppressionFile of suppressionFileList) {
+      let baseSuppressionContent = await getSdkSuppressionsFileContent(baseCommitHash, suppressionFile);
+      const headSuppressionContent = await getSdkSuppressionsFileContent(headCommitHash, suppressionFile);
+
+      // if the head suppression file is present but anything is wrong like schema error with it return
+      const validateSdkSuppressionsFileResult =
+        validateSdkSuppressionsFile(headSuppressionContent).result;
+      if (!validateSdkSuppressionsFileResult) {
+        return [];
+      }
+      // if base suppression file does not exist, set it to an empty object but has correct schema
+      if (!baseSuppressionContent) {
+        baseSuppressionContent = { suppressions: {} };
+      }
+
+      console.log(
+        `updateSdkSuppressionsLabels: Will compare base suppressions content:\n ` +
+        `${JSON.stringify(baseSuppressionContent)}\n ` + 
+        `and head suppressions content:\n ` +
+        `${JSON.stringify(headSuppressionContent)} to get different SDK.`,
+      );
+
+      sdkNameList = getSdkNamesWithChangedSuppressions(
+        headSuppressionContent as SdkSuppressionsYml,
+        baseSuppressionContent as SdkSuppressionsYml,
+      );
+    }
+  }
+
+  return [...new Set(sdkNameList)];
+}
+
+export async function getSdkSuppressionsFileContent(
+  ref: string,
+  path: string,
+): Promise<string | object | undefined | null> {
```

Comment:
```suggestion
): Promise<string | object | null> {
```

Reply:
This `undefined`  is a scenario when the yaml load run failed return in https://github.com/JackTn/azure-rest-api-specs/blob/b1d941811583667c9b8bb84109d240ea063f71cb/eng/tools/sdk-suppressions/src/common.ts#L19